### PR TITLE
Marking Matrix3x2, Matrix4x4, Plane, and Quaternion as Intrinsic

### LINKED ISSUE #41738: Performance regressions in Quaternion.Conjugate and Quaternion.Negate
It looks like `Conjugate` and `Negate` methods of the `Quaternion` type have regressed compared to 3.1.

@tannergooding could you please take a look and triage it? I assume that this might be acceptable similarly to  https://github.com/dotnet/runtime/issues/39035 but I don't have enough knowledge to make any decisions here.

## Repro

```cmd
git clone https://github.com/dotnet/performance.git
py .\performance\scripts\benchmarks_ci.py -f netcoreapp3.1 netcoreapp5.0 --filter 'System.Numerics.Tests.Perf_Quaternion.ConjugateBenchmark' 'System.Numerics.Tests.Perf_Quaternion.Negat*'
```

To see all the numbers please click "details" below:

<details>

## System.Numerics.Tests.Perf_Quaternion.ConjugateBenchmark

| Result |  Base |  Diff | Ratio | Alloc Delta | Modality | Operating System       | Bit   | Processor Name                              | Base V | Diff V      |
| ------ | -----:| -----:| -----:| -----------:| --------:| ---------------------- | ----- | ------------------------------------------- | ------ | ------------ |
| Slower |  1.30 |  7.91 |  0.16 |          +0 |          | Windows 10.0.18363.959 | X64   | Intel Xeon CPU E5-1650 v4 3.60GHz           | 3.1.6  | 5.0.20.41714|
| Slower |  6.43 | 18.76 |  0.34 |          +0 |          | manjaro                | X64   | Intel Core i7-4771 CPU 3.50GHz (Haswell)    | 3.1.6  | 5.0.20.41714|
| Slower | 12.90 | 21.39 |  0.60 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.6  | 5.0.20.41714|
| Slower | 12.90 | 27.30 |  0.47 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.7  | 5.0.20.41714|
| Slower | 14.44 | 24.09 |  0.60 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.6  | 5.0.20.41714|
| Slower |  1.00 |  7.67 |  0.13 |          +0 |          | Windows 10.0.18363.959 | X86   | Intel Xeon CPU E5-1650 v4 3.60GHz           | 3.1.6  | 5.0.20.41714|
| Faster |  9.44 |  5.20 |  1.81 |          +0 |          | Windows 10.0.19041.450 | Arm   | Microsoft SQ1 3.0 GHz                       | 3.1.6  | 5.0.20.41714|
| Slower |  8.01 | 23.80 |  0.34 |          +0 |          | macOS Mojave 10.14.5   | X64   | Intel Core i7-5557U CPU 3.10GHz (Broadwell) | 3.1.6  | 5.0.20.41714|

## System.Numerics.Tests.Perf_Quaternion.NegateBenchmark

| Result |  Base |  Diff | Ratio | Alloc Delta | Modality | Operating System       | Bit   | Processor Name                              | Base V | Diff V      |
| ------ | -----:| -----:| -----:| -----------:| -------- | ---------------------- | ----- | ------------------------------------------- | ------ | ------------ |
| Slower |  1.54 |  7.44 |  0.21 |          +0 |          | Windows 10.0.18363.959 | X64   | Intel Xeon CPU E5-1650 v4 3.60GHz           | 3.1.6  | 5.0.20.41714|
| Slower |  6.49 | 17.96 |  0.36 |          +0 |          | manjaro                | X64   | Intel Core i7-4771 CPU 3.50GHz (Haswell)    | 3.1.6  | 5.0.20.41714|
| Slower | 13.67 | 22.12 |  0.62 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.6  | 5.0.20.41714|
| Slower | 12.52 | 20.97 |  0.60 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.7  | 5.0.20.41714|
| Slower | 14.06 | 18.74 |  0.75 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.6  | 5.0.20.41714|
| Slower |  1.26 |  7.64 |  0.16 |          +0 | bimodal  | Windows 10.0.18363.959 | X86   | Intel Xeon CPU E5-1650 v4 3.60GHz           | 3.1.6  | 5.0.20.41714|
| Faster |  9.52 |  5.05 |  1.89 |          +0 |          | Windows 10.0.19041.450 | Arm   | Microsoft SQ1 3.0 GHz                       | 3.1.6  | 5.0.20.41714|
| Slower |  7.99 | 23.96 |  0.33 |          +0 |          | macOS Mojave 10.14.5   | X64   | Intel Core i7-5557U CPU 3.10GHz (Broadwell) | 3.1.6  | 5.0.20.41714|

## System.Numerics.Tests.Perf_Quaternion.NegationOperatorBenchmark

| Result |  Base |  Diff | Ratio | Alloc Delta | Modality | Operating System       | Bit   | Processor Name                              | Base V | Diff V      |
| ------ | -----:| -----:| -----:| -----------:| --------:| ---------------------- | ----- | ------------------------------------------- | ------ | ------------ |
| Slower |  1.56 |  7.47 |  0.21 |          +0 |          | Windows 10.0.18363.959 | X64   | Intel Xeon CPU E5-1650 v4 3.60GHz           | 3.1.6  | 5.0.20.41714|
| Slower |  6.48 | 18.24 |  0.36 |          +0 |          | manjaro                | X64   | Intel Core i7-4771 CPU 3.50GHz (Haswell)    | 3.1.6  | 5.0.20.41714|
| Slower | 12.90 | 22.14 |  0.58 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.6  | 5.0.20.41714|
| Slower | 12.90 | 22.12 |  0.58 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.7  | 5.0.20.41714|
| Slower | 13.67 | 24.44 |  0.56 |          +0 |          | ubuntu 16.04           | Arm64 | Unknown processor                           | 3.1.6  | 5.0.20.41714|
| Slower |  1.26 |  7.16 |  0.18 |          +0 |          | Windows 10.0.18363.959 | X86   | Intel Xeon CPU E5-1650 v4 3.60GHz           | 3.1.6  | 5.0.20.41714|
| Faster |  9.60 |  5.15 |  1.86 |          +0 |          | Windows 10.0.19041.450 | Arm   | Microsoft SQ1 3.0 GHz                       | 3.1.6  | 5.0.20.41714|
| Slower |  7.98 | 21.66 |  0.37 |          +0 |          | macOS Mojave 10.14.5   | X64   | Intel Core i7-5557U CPU 3.10GHz (Broadwell) | 3.1.6  | 5.0.20.41714|

</details>

@DrewScoggins this regression did not get detected by the bot as it was added very recently 👍

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #41738 by marking `Matrix3x2`, `Matrix4x4`, `Plane`, and `Quaternion` as intrinsic causing the `JitInlineSIMDMultipler` to be applied as it was for .NET Core 3.1

### CODE REVIEW

**Comment by tannergooding** [comment]
```
[2020/09/03 14:22:22][INFO] // * Summary *
[2020/09/03 14:22:22][INFO]
[2020/09/03 14:22:22][INFO] BenchmarkDotNet=v0.12.1.1405-nightly, OS=Windows 10.0.19041.388 (2004/May2020Update/20H1)
[2020/09/03 14:22:22][INFO] AMD Ryzen 9 3900X, 1 CPU, 24 logical and 12 physical cores
[2020/09/03 14:22:22][INFO] .NET Core SDK=6.0.100-alpha.1.20453.5
[2020/09/03 14:22:22][INFO]   [Host]     : .NET Core 5.0.0 (CoreCLR 5.0.20.41714, CoreFX 5.0.20.41714), X64 RyuJIT
[2020/09/03 14:22:22][INFO]   Job-EMLNHA : .NET Core 5.0 (CoreCLR 42.42.42.42424, CoreFX 42.42.42.42424), X64 RyuJIT
[2020/09/03 14:22:22][INFO]
[2020/09/03 14:22:22][INFO] PowerPlanMode=00000000-0000-0000-0000-000000000000  Arguments=/p:DebugType=portable  Toolchain=CoreRun
[2020/09/03 14:22:22][INFO] IterationTime=250.0000 ms  MaxIterationCount=20  MinIterationCount=15
[2020/09/03 14:22:22][INFO] WarmupCount=1
[2020/09/03 14:22:22][INFO]
[2020/09/03 14:22:22][INFO] |             Method |     Mean |     Error |    StdDev |   Median |       Min |      Max | Code Size | Gen 0 | Gen 1 | Gen 2 | Allocated |
[2020/09/03 14:22:22][INFO] |------------------- |---------:|----------:|----------:|---------:|----------:|---------:|----------:|------:|------:|------:|----------:|
[2020/09/03 14:22:22][INFO] | ConjugateBenchmark | 1.030 ns | 0.0309 ns | 0.0289 ns | 1.021 ns | 0.9983 ns | 1.104 ns |     130 B |     - |     - |     - |         - |
[2020/09/03 14:22:22][INFO] |    NegateBenchmark | 1.215 ns | 0.0165 ns | 0.0147 ns | 1.208 ns | 1.1971 ns | 1.239 ns |     142 B |     - |     - |     - |         - |
[2020/09/03 14:22:22][INFO]
```

**Review by pgovind — APPROVED** [review]
LGTM. You might want to tag the perf folks/Andy as FYIs

**Comment by CarolEidt** [comment]
Thanks @tannergooding - should this be considered for backporting?

**Review by jeffhandley — APPROVED** [review]
LGTM; thanks for taking care of this.  Once we have sign-off and this is merged into master, let's submit it for Ask Mode approval.

1. Use the port bot to port this to `release/5.0-rc2`
2. Fill out the PR template in the PR created by the bot, including notes for how this was caught, showing the regression numbers, and capturing the perf numbers you show here
3. Label it with `Servicing-consider`

Once we have that, I'll email Tactics to request approval.

**Comment by tannergooding** [comment]
Can you DM me instructions for using port bot?

**Comment by jeffhandley** [comment]
> Can you DM me instructions for using port bot?

Here's an example: https://github.com/dotnet/runtime/pull/41817#issuecomment-686640827

**Comment by AndyAyersMS** [comment]
@tannergooding can you run diffs? I know it's a bit clunky with libraries changes, but I am curious what impact this has.

**Comment by tannergooding** [comment]
>  can you run diffs? I know it's a bit clunky with libraries changes, but I am curious what impact this has.

Sure thing. It will take a bit to run, but I expect most changes will only be caught in our tests/benchmarks as we don't have much product code using these types.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you both @tannergooding  and @AndyAyersMS for identifying and solving the regression!

**Comment by tannergooding** [comment]
@AndyAyersMS, it was about as I expected with no visible diffs for anything except for the MicroBenchmarks that are covering the regression:
--frameworks
```
Found 802 files with textual diffs.

Summary of Code Size diffs:
(Lower is better)

Total bytes of diff: 0 (0.00% of base)

0 total files with Code Size differences (0 improved, 0 regressed), 267 unchanged.

0 total methods with Code Size differences (0 improved, 0 regressed), 259483 unchanged.

1 files had text diffs but no metric diffs.
diff\System.Private.CoreLib.dasm had 74 diff
```

--tests
```
Found 10519 files with textual diffs.

Summary of Code Size diffs:
(Lower is better)

Total bytes of diff: 0 (0.00% of base)

0 total files with Code Size differences (0 improved, 0 regressed), 3485 unchanged.

0 total methods with Code Size differences (0 improved, 0 regressed), 477140 unchanged.

64 files had text diffs but no metric diffs.
diff\JIT\Methodical\JIT.Methodical.XUnitWrapper.dasm had 133525 diffs
diff\JIT\Regression\JIT.Regression.XUnitWrapper.dasm had 108850 diffs
diff\JIT\HardwareIntrinsics\JIT.HardwareIntrinsics.XUnitWrapper.dasm had 61600 diffs
diff\JIT\Directed\JIT.Directed.XUnitWrapper.dasm had 24150 diffs
diff\JIT\IL_Conformance\JIT.IL_Conformance.XUnitWrapper.dasm had 19600 diffs
diff\JIT\jit64\JIT.jit64.XUnitWrapper.dasm had 18375 diffs
diff\JIT\SIMD\JIT.SIMD.XUnitWrapper.dasm had 16975 diffs
diff\Interop\PInvoke\Interop.PInvoke.XUnitWrapper.dasm had 15050 diffs
diff\JIT\Performance\JIT.Performance.XUnitWrapper.dasm had 14875 diffs
diff\JIT\opt\JIT.opt.XUnitWrapper.dasm had 11900 diffs
diff\JIT\Generics\JIT.Generics.XUnitWrapper.dasm had 7175 diffs
diff\JIT\CodeGenBringUpTests\JIT.CodeGenBringUpTests.XUnitWrapper.dasm had 5600 diffs
diff\JIT\Intrinsics\JIT.Intrinsics.XUnitWrapper.dasm had 4025 diffs
diff\Interop\COM\Interop.COM.XUnitWrapper.dasm had 3500 diffs
diff\Interop\IJW\Interop.IJW.XUnitWrapper.dasm had 1050 diffs
diff\Interop\StructMarshalling\Interop.StructMarshalling.XUnitWrapper.dasm had 875 diffs
diff\JIT\Stress\JIT.Stress.XUnitWrapper.dasm had 875 diffs
diff\Interop\NativeLibrary\Interop.NativeLibrary.XUnitWrapper.dasm had 700 diffs
diff\Interop\StringMarshalling\Interop.StringMarshalling.XUnitWrapper.dasm had 700 diffs
diff\Interop\ICustomMarshaler\Interop.ICustomMarshaler.XUnitWrapper.dasm had 525 diffs
```

--benchmarks
```
Found 246 files with textual diffs.

Summary of Code Size diffs:
(Lower is better)

Total bytes of diff: 0 (0.00% of base)

0 total files with Code Size differences (0 improved, 0 regressed), 82 unchanged.

0 total methods with Code Size differences (0 improved, 0 regressed), 1891 unchanged.
```

MicroBenchmarks.dll
```
Found 4 files with textual diffs.

Summary of Code Size diffs:
(Lower is better)

Total bytes of diff: 171 (0.01% of base)
    diff is a regression.

Top file regressions (bytes):
         171 : diff\MicroBenchmarks.dasm (0.01% of base)

1 total files with Code Size differences (0 improved, 1 regressed), 0 unchanged.

Top method regressions (bytes):
          61 (75.31% of base) : diff\MicroBenchmarks.dasm - Perf_Quaternion:NegationOperatorBenchmark():Quaternion:this
          61 (75.31% of base) : diff\MicroBenchmarks.dasm - Perf_Quaternion:NegateBenchmark():Quaternion:this
          49 (60.49% of base) : diff\MicroBenchmarks.dasm - Perf_Quaternion:ConjugateBenchmark():Quaternion:this

Top method regressions (percentages):
          61 (75.31% of base) : diff\MicroBenchmarks.dasm - Perf_Quaternion:NegationOperatorBenchmark():Quaternion:this
          61 (75.31% of base) : diff\MicroBenchmarks.dasm - Perf_Quaternion:NegateBenchmark():Quaternion:this
          49 (60.49% of base) : diff\MicroBenchmarks.dasm - Perf_Quaternion:ConjugateBenchmark():Quaternion:this

3 total methods with Code Size differences (0 improved, 3 regressed), 8326 unchanged.
```

**Comment by JulieLeeMSFT** [comment]
@tannergooding @AndyAyersMS Thanks a lot for rootcausing the issue and fixing it.

**Pull request merged**
